From 2fc1a03b8e763e8f5e3a868063962a0ad30f525f Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Wed, 1 Oct 2014 17:40:33 -0400
Subject: [ethernet] cxgb4: Fix t4_flash_erase_sectors() to throw an error when requested to erase sectors which aren't in the FLASH

Message-id: <1412185238-11567-3-git-send-email-svemuri@redhat.com>
Patchwork-id: 96828
O-Subject: [RHEL7.1 PATCH BZ 1147683 2/7] cxgb4: Fix t4_flash_erase_sectors() to throw an error when requested to erase sectors which aren't in the FLASH
Bugzilla: 1147683
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: David S. Miller <davem@redhat.com>

Upstream : https://git.kernel.org/cgit/linux/kernel/git/davem/net-next.git/commit/drivers/net/ethernet/chelsio?id=c0d5b8cf50c9842d0efcfb14871e58f42c6576e6

Based on original work by Casey Leedom <leedom@chelsio.com>

Signed-off-by: Hariprasad Shenai <hariprasad@chelsio.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit c0d5b8cf50c9842d0efcfb14871e58f42c6576e6)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c b/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
index 451c2d9..0501ea2 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
@@ -1100,6 +1100,9 @@ static int t4_flash_erase_sectors(struct adapter *adapter, int start, int end)
 {
  int ret = 0;
 
+ if (end >= adapter->params.sf_nsec)
+  return -EINVAL;
+
  while (start <= end) {
   if ((ret = sf1_write(adapter, 1, 0, 1, SF_WR_ENABLE)) != 0 ||
       (ret = sf1_write(adapter, 4, 0, 1,
-- 
1.7.1